armi/materials/material.py

```diff
@@ -240,6 +240,37 @@ def setDefaultMassFracs(self):
         pass
 
     def setMassFrac(self, nucName: str, massFrac: float) -> None:
+        """
+        Assigns the mass fraction of a nuclide within the material.
+
+        Notes
+        -----
+        This will try to convert the provided ``massFrac`` into a float
+        for assignment. If the conversion cannot occur then an error
+        will be thrown.
+        """
+        try:
+            massFrac = float(massFrac)
+        except Exception as ee:
+            raise TypeError(
+                f"Error is converting the mass fraction of {massFrac} "
+                f"for nuclide {nucName} in {self} to a float. "
+                f"Exception: {ee}"
+            )
+        if massFrac < 0.0:
+            raise ValueError(
+                f"Cannot assign a negative mass "
+                f"fraction of {massFrac} to {nucName} "
+                f"within {self}"
```

Comment:
I think this `{self}` will print something hideous.  Have you tried it, to know you like it?

You could also try: `{self.__class__.__name__}`, though I'm not sure f-string formatting would allow that, that's the prettier version for the material class.

Reply:
I think that `self` will print:

![image](https://user-images.githubusercontent.com/26268658/221324716-27e05007-8ed7-4eda-8e93-a05243bb9495.png)